src/v1/sources/shopify/webpixelTransformations/pixelTransform.js

```diff
@@ -147,6 +148,36 @@ function processPixelEvent(inputEvent) {
   });
   message.setProperty('context.topic', name);
   message.setProperty('context.shopifyDetails', shopifyDetails);
+
+  // adding campaign object to the message
+  if (context?.document?.location?.href) {
+    const url = new URL(context.document.location.href);
+    const campaignParams = {};
+
+    // Loop through mappings and extract UTM parameters
+    campaignObjectMappings.forEach((mapping) => {
+      const value = url.searchParams.get(mapping.sourceKeys);
+      if (value) {
+        campaignParams[mapping.destKeys] = value;
+      }
+    });
+
+    // Extract any UTM parameters not in the mappings
+    const campaignObjectSourceKeys = campaignObjectMappings.flatMap(
+      (mapping) => mapping.sourceKeys,
+    );
+    url.searchParams.forEach((value, key) => {
```

Comment:
Is it not forEach((key, value) ?

Reply:
no, campaignParams[key] will get campaign { } keys, and value will be the utm values in the href